### PR TITLE
Upgrade latest gradle and sdk versions. Fix build error because of:   classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.1.1'

### PR DESCRIPTION
**Pull request by IgorGanapolsky**
Upgrade latest gradle and sdk versions.
Fix build error because of:
  classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.1.1'

### CODE REVIEW

**Comment by jasta** [comment]
Why do we want this patch?  It upgrades to the latest build tools, but for what reason?  Also, it breaks our automated tests, which of course is a non-starter...

**Comment by IgorGanapolsky** [comment]
Latest build tools and latest SDK is what many developers have installed on
their machines.  If you resist such changes and such progress, you are
setting your project up for friction with dev community.

Igor

**Comment by jasta** [comment]
@IgorGanapolsky of course, that makes sense.  But we still need our tests working, of course :)  Any chance you can follow up as to why Robolectric testing is not working properly with the latest build tools?  AFAICT, they recommend using the plugin that you had to disable.

**Comment by IgorGanapolsky** [comment]
Well, jcandksolutions shouldn't be used anymore since Gradle now supports
jvm testing in projects natively.  In fact, if you check jcandksolutions
say on his github repo that his project is deprecated.

Igor

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 *.iml
 .DS_Store
 /build
+/.project
```
Comment (jasta) [inline .gitignore]: What is this for?

**Comment by jasta** [comment]
Oh I didn't realize that was tested to work with Robolectric as well.  Looks like the Robolectric sample was updated accordingly as well: https://github.com/robolectric/deckard-gradle/blob/master/build.gradle

Though they don't use gradle 1.3.0 yet.  Can you please test that `gradle check connectedCheck` still runs all the tests?  Also note that Travis CI doesn't appear to have the latest build tools available for some reason.  We won't be able to merge without that being resolved.

File: build.gradle [hunk 2]
```diff
@@ -3,8 +3,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.0.0'
-        classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.1.1'
+        classpath 'com.android.tools.build:gradle:1.1.3'
+//        classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.1.1'
```
Comment (jasta) [inline build.gradle]: If this is truly no longer needed, please just delete the line.

File: stetho-okhttp/build.gradle [hunk 3]
```diff
@@ -20,7 +20,7 @@ android {
     }
 }
 
-apply plugin: 'android-unit-test'
+//apply plugin: 'android-unit-test'
```
Comment (jasta) [inline stetho-okhttp/build.gradle]: Remove these as well.

File: .gitignore (hunk 1 above)
Comment (IgorGanapolsky) [inline .gitignore]: As far as .gitignore file, that was added by my GitEye client
automatically.  I'll look into it.

**Comment by fourlastor** [comment]
It might help to change the script entry in the travis config to `./gradlew test --continue`.
This runs the unit tests and it worked for my company to run CI on travis and the last gradle plugin.

More info here: http://tools.android.com/tech-docs/unit-testing-support

**Comment by jasta** [comment]
@fourlastor without tests working in Travis I'm inclined to just wait for them to fix this upstream and then integrate your patch.  As of now it doesn't seem the value is worth the cost of a loss of tests in CI.

**Comment by fourlastor** [comment]
What I'm saying is that you don't need to run the `connectedCheck` task to run the tests: stetho has only unit tests so the `test` task will do just fine.

The only test that doesn't pass is `ObjectMapperTest#testNestedProperty` because the `JSONObject#toString` method returns the keys in a different order, but the problem there is asserting equality on `JSONObject`s with the toString method.

All the suite of test (14 in the stetho project, 2 in the stetho-okhttp project) passed for me with this setup:

- Upgrade gradle plugin to 1.1.0
- Remove ` 'com.github.jcandksolutions.gradle:android-unit-test:2.1.1'` from the main build.gradle
- Remove `apply plugin: 'android-unit-test'` from the projects build.gradle

For the `JSONObject#toString` problem I suggest using an assertion library meant for it (such as https://github.com/skyscreamer/JSONassert).

**Comment by jasta** [comment]
@fourlastor Ack'd.  Curious though why the removal of `android-unit-test` would break a flaky test that was previously working.  Doesn't matter though, we can just fix the test.

As for Travis CI, there remains the more serious issue of build tools 22.0.0 being apparently missing (check the Travis CI build output).  Until that passes, we cannot merge your change.

**Comment by fourlastor** [comment]
It might be that it's not working on *my* machine, due to the JDK I'm using or whatever.

It's not my changes :P I'm not the PR author.

I would suggest to split this in 2 separate PR tho:
- unit test plugin removal (and bump of the Gradle plugin version to 1.1.x), which works and it's great to have
- bump of build tools to 22

**Comment by fourlastor** [comment]
Travis CI is going to update their environments tomorrow, including Android SDK 22:

http://docs.travis-ci.com/user/build-environment-updates/2015-04-09/

**Comment by jasta** [comment]
This is being divided and worked on as separate PRs.

**Pull request closed without merging**